docs(shadows): add documentation about usage of our shadow variables … (#1226)
Lundalogik/lime-elements: src/shadows-and-depth.md

> # Shadows and depth

Reviewer: Kiarokh wrote:
> Do we have to say anything about how they can import these css variables? Or is it a given that they know?

Not in this guide at least. For those building add-ons, they won't have to import these at all, since that's done by the client. We might want to document how to do _that_ somewhere, but I don't think that belongs in this design guide.

If this is your only intention, we recommend using a different background color, use different font characteristics, or rely on a good layout design that uses headings, paddings and margins well.
:::
<limel-example-shadows-bad-usage></limel-example-shadows-bad-usage>

Sometimes, using the recommended techniques above does not suffice, when you want to visually emphasize on a section of the UI and make it pop out. For such cases, we offer other variables which can be used as `box-shadow` on the element.

<limel-example-surface-shadows-inflated></limel-example-surface-shadows-inflated>

:::tip
The best visual expression with these variables is achieved when:
- the element has a sizable `border-radius`
- there is enough of empty space around the element
- the element has a light grey as `background-color`, (preferably `var(--contrast-200)`, `var(--contrast-300)` or `var(--contrast-400)`)
- `background-color` of the surface below the element is not too dark or too light (preferably `var(--contrast-400)`, `var(--contrast-500)` or `var(--contrast-600)`)
- the element's `background-color` is lighter than the surface below it
:::

:::warning

> - Do not over use this visual effect on the same screen.

Reviewer: ```suggestion
- Do not overuse this visual effect on the same screen.
```

…but I had to read this twice to understand what you meant by "on the same screen". I understand what you mean, and I'm having a bit of trouble coming up with a better word, but we don't really use or mention the concept of "screens" in that sense anywhere else, so I'd prefer some alternative here. Perhaps rewording a bit might help. Something like:

```suggestion
- Do not use this effect on too many elements visible at the same time. If all elements are shouting for attention, the result is just a noisy interface.
```

> - The element using this effect should not be too small. Because the effect may make it look like a clickable button.

Reviewer: ```suggestion
- The element using this effect should not be too small, or the effect might make it look like a clickable button.
```

I also don't know if "clickable" needs to be there. Buttons are clickable, that's their main feature.

So perhaps this, if you like it:

```suggestion
- The element using this effect should not be too small, or the effect might make it look like a button.
```

# Shadows and depth

## Physical world metaphores in UI design

> We who perceive and navigate a digital product using our vision, can better and faster make sense of the user interface when elements of the real world are used in its design. Some have called this Skeuomorphism in user interface design, where the designer tries to emulate the visual qualities of physical objects, in the digital 2 dimensional interface. A good example of an extreme skeuomorphic design is the first calendar app on the iPhone. Its UI design featured visual details such as leather cover and paper details, to mimic a physical pocket calendar.

Author: perfect. i'll use

> We who perceive and navigate a digital product using our vision, can better and faster make sense of the user interface when elements of the real world are used in its design. Some have called this Skeuomorphism in user interface design, where the designer tries to emulate the visual qualities of physical objects, in the digital 2 dimensional interface. A good example of an extreme skeuomorphic design is the first calendar app on the iPhone. Its UI design featured visual details such as leather cover and paper details, to mimic a physical pocket calendar.

Author: used this

> Despite skeuomorphism having already been replaced with more minimalistic UI design trends, we still use details from the real world in our UI designs, to help users make sense of the elements of digital user interfaces. From a shutter-click sound of a camera app on the phone, to elements such as toggle buttons, to notions such as light, shadows, and motion.

Author: agree :)

> We may look at a delicate visual clue such as an elevated button with a slight shadow underneath it, and semi-consciously perceive it as an interactable element of the user interface. This is because the elevation reminds us of a real physical button. Just by looking at it we can realize that this thing can afford to be pressed down, pushed, or to the slid to the sides. We may see a toolbar or a modal placed on top of the rest of the elements of the screen, casting a slight shadow on whatever is below it. This helps us realize that the element has a higher hierarchy, is more important, and or demands more attention right now.

Author: done

## Shadows
When correctly used, shadows can improve usability of UI elements, for users who rely on their vision to navigate and use the interface. We can use shadows to:
- create a sense of depth,
- improve perception of users of clickable elements,
- emphasize on the hierarchy of sections of the user interface,
- help users understand temporariness of a component that overlays itself on top of a context,
- clarify direction of the movement of an element, whether it is animated as a result of a user interaction such as pressing a button, or it is animated - to indicate transition from one state to another,
- and even visualize distinct surfaces by clarifying their edges.

A surface’s shadow is determined by its elevation and relationship to other surfaces.

> Because shadows express the degree of elevation between surfaces, they must be used consistently throughout your product.

Author: I'll change but are we not trying to be an open source design system. And offer components that other products can also use? :) So I think we can say "your". But "the" works well too. ;)

## CSS variables that we offer for shadows
Because shadows express the degree of elevation between surfaces, they must be used consistently throughout your designs. Therefore, Lime Elements offers a handfull of CSS variable that can be used as `box-shadow` properties on desired elements.

:::tip
Using these variables also ensures that on dark mode, shadows are automatically rendered darker to be more visible on dimmer backgrounds.
:::

### Shadows variables for buttons

> Buttons or similar clickable/tappable elements such as dropdown triggers or slider handels particularly benefit from the visual effect caused by shadows. Because a shadow gives the element an affordance to be pressed, since it is elevated.

Author: I use slash to say either this or that. But Agree with you.

"Affordance" is a design term. But good that you changed it.

#### Change interactive shadows smoothly
It's crucial that an interaction (like hovering or pressing) that changes a visual property of an element such as its depth, is done smoothly. In other words change of shadows should not happen instantly. It should rather be animated with a transition. So make sure you have specified a `transition` property for `box-shadow`, like below.

````scss
transition: box-shadow 0.2s ease;
````

## When not to use shadows/depth
Sometimes we may only be trying to use shadows to create a visually distinct section, to separate a part of the interface from other parts. In theory, having visually distinct areas in the UI increases the ease and speed of finding components or information when users are quickly scanning pages. But this can be done through other ways as well, such as simply using outlines, dividers, or different background colors.

:::warning
We want to spare the visual power of shadows, to improves one’s ability to determine whether or not something can be interacted with, or has a higher hierarchy within the context. Therefore, we strongly discourage usage of shadows with the only purpose of creating a visually distinct section in the UI.

> If that is your only intention, we recommend using a different background color, use different font characteristics, or rely on a good layout design that uses paddings and margins well.

Author: I don't understand your question
